python/pyproject.toml

```diff
@@ -86,6 +88,8 @@ pyarrow = ["pyarrow"]
 snappy = ["python-snappy"]
 hive = ["thrift"]
 s3fs = ["s3fs"]
+boto3 = ["boto3"]
```

Comment:
How about naming this `glue` instead? This makes it easier when installing extras: `pip install pyiceberg[glue]` instead of `pip install pyiceberg[boto3]`, because people are interested in glue, and don't care about boto3.

Reply:
```suggestion
glue = ["boto3"]
```